fix: update defaults for mlflow logging

anaprietonem opened the pull request:
## Description
<!-- What issue or task does this change relate to? -->
PR to update the default settings regarding tracking of system metrics. New default is to set those to `False` so that if the wants to track them, they will need to activate the system metrics by updating this to True. 

## What problem does this change solve?
<!-- Describe if it's a bugfix, new feature, doc update, or breaking change -->
Default interval for logging system metrics was every 10 seconds. This lead to a large amount of information being logged into the mlflow server. 


## What issue or task does this change relate to?
<!-- link to Issue Number -->
https://github.com/ecmwf/anemoi-core/issues/332

##  Additional notes ##
<!-- Include any additional information, caveats, or considerations that the reviewer should be aware of. -->

***As a contributor to the Anemoi framework, please ensure that your changes include unit tests, updates to any affected dependencies and documentation, and have been tested in a parallel setting  (i.e., with multiple GPUs). As a reviewer, you are also responsible for verifying these aspects and requesting changes if they are not adequately addressed. For guidelines about those please refer to https://anemoi.readthedocs.io/en/latest/***

gmertes reviewed (commented):
Nice, LGTM. Can you rename the PR (chore is ignored) so that it gets included in the release notes. Maybe `feat(config)` ?

anaprietonem commented:
@gmertes  I renamed the PR to 'fix'. And following up from our conversation, I have simplified the logic so that the interval is kept to 100 second for system metrics and the default is not to log them

gmertes reviewed (approved):
Thanks! Should reduce the logging overhead significantly.
